content/blog/2017/2017-01-17-conditional.adoc

```diff
@@ -0,0 +1,464 @@
+but migrating jobs using complex conditional build steps will require a bit more care.
+
+
+== Graphical Programming
+
+The Conditional BuildStep plugin lets users add conditional logic to Freestyle
+jobs from within the Jenkin UI.  It does this by:
+
+* Adding two types of Conditional BuildStep ("Single" and "Multiple") -
+these build steps contain one or more other build steps to be run when the configured
+condition is met
+* Adding a set of Condition operations -
+these control whether the Conditional BuildStep execute the contained step(s)
+* Leveraging the Token Macro facility -
+these provide values to the Conditions for evaluation
+
+In the example below, this project will run the shell script step when the value of the
+`REQUESTED_ACTION` token equals "greeting".
+
+image::/images/post-images/2017-01-17/freestyle-conditional-param.png[Freestyle Job Parameters, role="center", width=800]
+
+image::/images/post-images/2017-01-17/freestyle-conditional-config.png[Freestyle Job Conditional BuildStep, role="center", width=800]
+
+Here's the output when I run this project with `REQUESTED_ACTION` set to "greeting":
+
+[source]
+----
+Run condition [Strings match] enabling prebuild for step [Execute shell]
+Strings match run condition: string 1=[greeting], string 2=[greeting]
+Run condition [Strings match] enabling perform for step [Execute shell]
+[freestyle-conditional] $ /bin/sh -xe /var/folders/hp/f7yc_mwj2tq1hmbv_5n10v2c0000gn/T/hudson5963233933358491209.sh
++ echo 'Hello, bitwiseman!'
+Hello, bitwiseman!
+Finished: SUCCESS
+----
+
+And when I pass the value "silence":
+
+[source]
+----
+Run condition [Strings match] enabling prebuild for step [Execute shell]
+Strings match run condition: string 1=[greeting], string 2=[greeting]
+Run condition [Strings match] enabling perform for step [Execute shell]
+[freestyle-conditional] $ /bin/sh -xe /var/folders/hp/f7yc_mwj2tq1hmbv_5n10v2c0000gn/T/hudson5963233933358491209.sh
++ echo 'Hello, bitwiseman!'
+Hello, bitwiseman!
+Finished: SUCCESS
+----
+
+This is a simple example but the conditional step can contain any regular build step.
+When combined with other plugins, it can control whether to send notifications,
+gather data from other sources, wait for user feedback, or call other projects.
+
+The Conditional BuildStep plugin does a great job of leveraging strengths of
+the Jenkins UI, and freestyle jobs, and UI-based programming,
+but it is also hampered by their limitations.
+The Jenkins UI can be clunky and confusing at times.
+Like the steps in any freestyle job the logic in these conditional steps is only
+stored and viewable in Jenkins.
+It is not versioned with other product or build code and can't be code reviewed.
+Like any number of UI-based programming it has to make trade-offs between clarity
+and flexibility: more options or clearer presentation.
+There's only so much space on the screen.
+
+== Migrating to Pipeline
+
+The Pipeline equivalent of the example above is:
```

Comment:
Yeah, I had a TODO for a transitional paragraph. 
I thought I might be able to get by without it, but no. 😄